### PR TITLE
Proposed cleanup, move prod build to pipeline, make commands more con…

### PR DESCRIPTION
**Pull request by asomervell**

### CODE REVIEW

**Comment by asomervell** [comment]
@umbra268 next config should probably overtly put the output of the prod build into a /dist folder the docker container can pick up easily?

**Comment by umbra268** [comment]
next build does output a production optimised build into the .next folder.
though it does require the rest of the files to execute the server part.

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -193,6 +193,14 @@ jobs:
           ${{ runner.os }}-build-
           ${{ runner.os }}-  
 
+    - name: Build production
+      run: |
+        npm ci --production
+        npm run build:production
+      env:
+        ENV_ENVIRONMENT: beta
+        ENV_SECRET: ${{ secrets.ENV_SECRET }}
```
Comment (umbra268) [inline .github/workflows/push.yml:202]: 👍

File: package.json [hunk 2]
```diff
@@ -10,27 +10,17 @@
     "test:components": "npx cross-env NODE_ENV=test PORT=8080 ENV_ENVIRONMENT='development' ENV_SECRET='mwmBqNcTWiKxDHygMiKhwyffaKQCVoRQ' nyc --reporter=lcov --reporter=text ava `find components/ -name \"*.spec.js\"`",
     "test:lib": "npx cross-env NODE_ENV=test PORT=8080 ENV_ENVIRONMENT='development' ENV_SECRET='mwmBqNcTWiKxDHygMiKhwyffaKQCVoRQ' nyc --reporter=lcov --reporter=text ava `find lib/ -name \"*.spec.js\"`",
     "watch:test": "npm run test -- --watch",
-    "end-to-end-test": "run-p -r dev testcafe-tests",
-    "testcafe-tests": "wait-on -t 60000 -w 2000 -l http://localhost:3122 && node node_modules/testcafe/bin/testcafe.js --skip-js-errors firefox:headless systemtest/landing.test.js",
-    "lint": "standard",
-    "fix": "standard --fix",
-    "dev": "cross-env ENV_ENVIRONMENT='development' ENV_SECRET='mwmBqNcTWiKxDHygMiKhwyffaKQCVoRQ' nodemon -w lang -w server -w package.json server/server.js",
```
Comment (umbra268) [inline package.json:17]: Would recommend keeping, lint, fix, & dev.

File: Dockerfile [hunk 3]
```diff
@@ -1,40 +1,20 @@
...
-ENV MONGOMS_VERSION="v4.0-latest"
-ENV ENV_ENVIRONMENT='development'
-ENV ENV_SECRET='mwmBqNcTWiKxDHygMiKhwyffaKQCVoRQ'
 ENV PORT 3122
 EXPOSE 3122
-
 COPY package.json package-lock.json* ./
 RUN npm install
 COPY . ./
-CMD ["npm", "run", "dev"]
```
Comment (umbra268) [inline Dockerfile:40]: Might be a good idea to split the prod and dev docker file.

As this should work for production, but we may need to modify the dev docker & docker-compose more heavily.

Comment (asomervell) [reply Dockerfile:40]: Yep for sure. Can do. Feel free to commit! :p

File: package.json (hunk 2 above)
Comment (asomervell) [reply package.json:17]: Oh ya I just moved them down I think?

Comment (avowkind) [reply package.json:17]: we can take out the testcafe-tests
I want to replace them with codecept tests

**Comment by avowkind** [comment]
I suggest closing this PR and restarting it - its somewhat behind current and there are likely to be conflicts in the docker stuff

**Pull request closed without merging**